### PR TITLE
Integrate WSAStartUp into picosocks

### PR DESCRIPTION
**Pull request by steschu77**
Add WSAStartUp to picosocks.c and remove indicidual initialization from picoquicdemo and tests.

That helps users of picoquic to avoid debugging Windows-related network issues.

### CODE REVIEW

File: picoquic/picosocks.c [hunk 1]
```diff
@@ -284,6 +284,8 @@ int picoquic_socket_set_ecn_options(SOCKET_TYPE sd, int af, int * recv_set, int
 SOCKET_TYPE picoquic_open_client_socket(int af)
 {
 #ifdef _WINDOWS
+    WSADATA wsaData = { 0 };
+    WSA_START(MAKEWORD(2, 2), &wsaData);
     SOCKET_TYPE sd = WSASocket(af, SOCK_DGRAM, IPPROTO_UDP, NULL, 0, WSA_FLAG_OVERLAPPED);
 #else
```
Comment (huitema) [inline picoquic/picosocks.c]: Looks good, but that means calling WSA_START several times, once per socket.

**Comment by huitema** [comment]
Looking at the [WSAStartUp page on MSDN](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsastartup) shows the potential issue with calling that function multiple times: _An application must call the WSACleanup function for every successful time the WSAStartup function is called._ I am not sure how big a deal that is in practice, because we never call WSACleanup. According to the MSDN page, that should be OK as long as all calls are for the same Winsock version.

**Comment by steschu77** [comment]
We can make sure the WSAStartUp is called really once using some global
variable. I know that is ugly, I just want to know your opinion :)

**Comment by huitema** [comment]
Yes, I know that there is an ugly solution. Let's just check the code as is, and wait for someone to complain about memory leak or whatever...

**Comment by steschu77** [comment]
Ok, I'll just add the double socket open test.

**Comment by steschu77** [comment]
@huitema please review the test and merge if it's ok.

**Pull request merged**